ros2_batch_job/__main__.py

```diff
@@ -553,19 +446,30 @@ def need_package_from_pipy(pkg_name):
                 ['"%s"' % job.python, '-m', 'pip', 'install', '-U'] +
                 ['colcon/%s' % pkgname for pkgname, importname, version in colcon_packages],
                 shell=True)
+            print('# END SUBSECTION')
 
         colcon_script = which('colcon')
 
-        # Show what pip has
-        job.run(['"%s"' % job.python, '-m', 'pip', 'freeze', '--all'], shell=True)
-        print('# END SUBSECTION')
-
         # Fetch colcon mixins
         if args.colcon_mixin_url:
+            print('# BEGIN SUBSECTION: Fetch colcon mixins')
             true_cmd = 'VER>NUL' if sys.platform == 'win32' else 'true'
             job.run([colcon_script, 'mixin', 'remove', 'default', '||', true_cmd], shell=True)
             job.run([colcon_script, 'mixin', 'add', 'default', args.colcon_mixin_url], shell=True)
             job.run([colcon_script, 'mixin', 'update', 'default'], shell=True)
+            print('# END SUBSECTION')
+
+        print('# BEGIN SUBSECTION: Print python versions')
+        # Print setuptools version
+        job.run(['"%s"' % job.python, '-c', '"import setuptools; print(setuptools.__version__)"'],
+                shell=True)
+
+        # Print the pip version
+        job.run(['"%s"' % job.python, '-m', 'pip', '--version'], shell=True)
+
+        # Show what pip has
+        job.run(['"%s"' % job.python, '-m', 'pip', 'list'], shell=True)
```

Comment:
I wonder if we wouldn't prefer `pip freeze --all` here in order to get information about all installed packages. I can't say exactly what's excluded without the `--all` but it is bothersome to me that some things are.

This suggestion is non-blocking.


```suggestion
        job.run(['"%s"' % job.python, '-m', 'pip', 'freeze', '--all'], shell=True)
```

Reply:
Right, this is a good question.  Early on in development here, I found that `pip freeze --all` just would not work properly in a Conda environment.  I seem to remember it giving me either empty or weird results.  Because of that, I ended up changing to `pip list` everywhere.

But it is long enough ago that I now don't remember exactly what the problem was.  Let me go back and do some testing here and see if that was either a misconfiguration, or something more serious here.

Reply:
Oh, right.  The issue popped up immediately when I tried it locally.

For reasons I don't understand, when installing python packages via pixi/conda, the output from `pip freeze --all` looks like this:

```
(pixi_ros2_rolling) C:\dev\rolling>pip freeze --all
argcomplete @ file:///home/conda/feedstock_root/build_artifacts/argcomplete_1698912039670/work
catkin-pkg @ file:///home/conda/feedstock_root/build_artifacts/catkin_pkg_1694651879177/work
cffi @ file:///D:/bld/cffi_1725560655177/work
...
```

(truncated for brevity).  As you can see, the output there doesn't give much useful information about which *version* of the package is installed, and thus in the CI job output it isn't very helpful.  That's why I switched to `pip list` everywhere.